includes/auto-distribute.php

```diff
@@ -0,0 +1,440 @@
+
+	if ( ! $post || ! $user_id ) {
+		return;
+	}
+
+	if ( ! in_array( $post->post_type, auto_distribute_supported_post_types(), true ) ) {
+		// If the post type is not supported, do not auto distribute.
+		return;
+	}
+
+	$connections = get_connections( $post->ID, $user_id );
+	if ( empty( $connections ) ) {
+		return;
+	}
+
+	foreach ( $connections as $connection ) {
+		if ( $connection['syndicated'] ) {
+			continue;
+		}
+
+		$connection_id   = $connection['id'];
+		$connection_type = $connection['type'];
+
+		distribute_post( $post->ID, $user_id, $connection_id, $connection_type );
+	}
+}
+
+/**
+ * Distribute a post to a specific connection.
+ *
+ * This function pushes the post to the specified connection and updates
+ * the connection map in the post meta.
+ *
+ * @param int    $post_id         Post ID.
+ * @param int    $user_id         User ID of the post author.
+ * @param int    $connection_id   Connection ID.
+ * @param string $connection_type Type of connection ('external' or 'internal').
+ */
+function distribute_post( $post_id = 0, $user_id = 0, $connection_id = 0, $connection_type = '' ) {
+	$post = get_post( $post_id );
+
+	if ( ! $post || ! $user_id || ! $connection_id || ! $connection_type ) {
+		return;
+	}
+
+	if ( ! auto_distribute_post( $post, $user_id, $connection_type, $connection_id ) ) {
+		// The post should not be auto-distributed, do not proceed.
+		return;
+	}
+
+	// Make sure we have a proper user set up
+	if ( ! $user_id ) {
+		$user_id = get_current_user_id();
+	}
+	wp_set_current_user( $user_id );
+
+	if ( 'external' === $connection_type ) {
+		$connection = \Distributor\ExternalConnection::instantiate( $connection_id );
+	} elseif ( 'internal' === $connection_type ) {
+		$connection = new \Distributor\InternalConnections\NetworkSiteConnection( get_site( $connection_id ) );
+	} else {
+		return; // Invalid connection type
+	}
+
+	$post_id        = $post->ID;
+	$connection_map = get_post_meta( intval( $post_id ), 'dt_connection_map', true );
+	if ( ! is_array( $connection_map ) ) {
+		$connection_map = [];
+	}
+	if ( empty( $connection_map['internal'] ) ) {
+		$connection_map['internal'] = [];
+	}
+	if ( empty( $connection_map['external'] ) ) {
+		$connection_map['external'] = [];
+	}
+	$args        = array(
+		'post_status' => default_post_status( $post ),
+	);
+	$remote_post = $connection->push( $post_id, $args );
+	if ( ! is_wp_error( $remote_post ) && ! empty( $remote_post['id'] ) ) {
+		// Store the connection mapping
+		$connection_map[ $connection_type ][ $connection_id ] = [
+			'post_id' => (int) $remote_post['id'],
+			'time'    => time(),
+		];
+
+		// Update the post meta with the new connection map
+		update_post_meta( intval( $post_id ), 'dt_connection_map', $connection_map );
+	}
+}
+
+/**
+ * Get all connections
+ *
+ * @param integer $post_id Post ID.
+ * @param integer $user_id User ID.
+ * @return array
+ */
+function get_connections( $post_id = 0, $user_id = 0 ) {
```

Comment:
Just wondering – do we already have some of these common functions in a helper class or utility file that we could leverage here? That might help keep things consistent and reusable.

Same question for the other functions added below, like `get_external_connections` and `get_internal_connections`.

Reply:
We have similar code in `Distributor\PushUI\get_connections()` that is behind a nonce and capability check, so not appropriate in this instance as it needs to run on cron.

Do you want me to see if I can share the functionality between this and the admin-ajax endpoint and move this to the utils namespace?

Reply:
Yeah, if you believe separating those common functions and moving them to the utils namespace would be beneficial for future enhancements (and it does not require too much efforts), then we should go ahead with that. Otherwise, I’m also fine keeping it as is for now; happy to go with your judgment on this.